### ISSUE TITLE
React component is reported as unused when used in JSX

### ISSUE THREAD

**Issue by pgilad** [eslint/eslint#4227]
I am getting `no-unused-vars` triggered when having an imported React component, and using it along with `React.render` with JSX Syntax.

Eslint `v1.6.0`

Reduced example:
```js
import LabelsDropdown from './labels-dropdown';

// somewhere along the code in same file:
const $elm = this.$('.labels-filter')[0];
React.render(<LabelsDropdown hideDropdown={ this.removeComponent.bind($elm) } />, $elm);
```

And eslint lints this with: `"LabelsDropdown" is defined but never used (no-unused-vars)`.

My `.eslintrc` config is:

```json
{
    "ecmaFeatures": {
        "modules": true,
        "jsx": true
    },
    "env": {
        "es6": true,
        "browser": true,
        "node": true,
        "jquery": true,
        "mocha": true
    },
    "extends": "eslint:recommended",
    "rules": {
        "arrow-parens": [1, "as-needed"],
        "arrow-spacing": [1, { "before": true, "after": true }],
        "brace-style": [2, "1tbs"],
        "comma-dangle": [0, "always-multiline"],
        "curly": [2, "all"],
        "dot-notation": [1, { "allowKeywords": false, "allowPattern": "" }],
        "jsx-quotes": [1, "prefer-double"],
        "no-const-assign": [2],
        "no-unused-vars": [1, {"vars": "all", "args": "after-used"}],
        "no-use-before-define": [1],
        "prefer-const": [1],
        "prefer-spread": [1],
        "prefer-template": [0],
        "quotes": [1, "single"],
        "yoda": [1, "never"],
        "semi": [1, "always"],
        "semi-spacing": [1, {"before": false, "after": true}]
    }
}
```

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
This is React-specific syntax. You need to use [eslint-plugin-react](https://github.com/yannickcr/eslint-plugin-react)

**Comment by pgilad** [comment]
Yep that solved the issue. Thanks for the help!

**Comment by just-boris** [comment]
Why it is needed an extra plugin for `React`, but JSX is supported from the box?
JSX can't be used without React, so why not to put JSX support and React-specific rules under a single configuration option?

**Comment by lo1tuma** [comment]
@just-boris that's not correct. JSX can and actually is used without react.
Examples: [msx](https://github.com/insin/msx), [mercury-jsx](https://github.com/Raynos/mercury-jsx), [jsx-dom](https://github.com/vjeux/jsxdom).

**Comment by just-boris** [comment]
Well, but look at this from my side: I am using React, I see the `jsx` support and I guessed that is enough to enable React support for my project.

Maybe it should add a related warning near the `jsx` option, something like "If you are going to use JSX, don't forget to add a framework-specific plugin to get proper support". I can send a pull request with it.

**Comment by lo1tuma** [comment]
That’s already [mentioned in our README.md](https://github.com/eslint/eslint#does-eslint-support-jsx).

**Comment by just-boris** [comment]
At the bottom of file :)

Ok, I got this. It can be as is, unless more people come to the issue.
